File: Day116_2025/Q116_Leetcode.py
from typing import List


class Solution:
    def countSubarrays(self, nums: List[int], minK: int, maxK: int) -> int:
        ans = 0
        minPos = maxPos = leftBound = -1
        
        for i, num in enumerate(nums):
            if num < minK or num > maxK:
                leftBound = i
            if num == minK:
                minPos = i
            if num == maxK:
                maxPos = i
            count = min(minPos, maxPos) - leftBound
            ans += 0 if count <= 0 else count
        
        return ans

File: Day116_2025/Q116_Leetcode.java
class Q116_Leetcode {
    public long countSubarrays(int[] nums, int minK, int maxK) {
        long ans = 0;
        int minPos = -1, maxPos = -1, leftBound = -1;
        
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] < minK || nums[i] > maxK) {
                leftBound = i;
            }
            if (nums[i] == minK) {
                minPos = i;
            }
            if (nums[i] == maxK) {
                maxPos = i;
            }
            int count = Math.min(minPos, maxPos) - leftBound;
            ans += (count <= 0) ? 0 : count;
        }
        
        return ans;
    }
}
